swap/src/database.rs

```diff
@@ -134,6 +172,12 @@ impl Database {
             Ok((swap_id, swap.try_into_bob()?))
         })
     }
+
+    pub fn unfinished_alice(&self) -> Result<Vec<(Uuid, Alice)>> {
+        self.all_iter_alice()
+            .filter_ok(|(_swap_id, alice)| !matches!(alice, Alice::Done(_)))
+            .collect()
+    }
```

Comment:
This adds complexity on the caller side, because we have an async call when sending the swap through the channel. Additionally I find it more readable in the current state. Won't optimize for now :)

Reply:
Sure! I don't quite understand why it adds complexity, there is virtually no difference in iterating over a `Vec` vs an `Iterator` except that the latter evaluates lazily and if you already have an `Iterator` here, you save yourself an allocation by not calling `collect` on it.

Reply:
When I gave it a try I used combinators on the outside to operate on the iterator. We have an async call to fetch the peer ID for each swap - which was then done within the combinator and that caused problems. Not sure why, maybe it was just me :)